beat/heart.py

```diff
@@ -3281,11 +3355,24 @@ def gamma(takeoffangles, azimuths):
     sa = num.sin(azimuths)
     return num.array([st*ca, st*sa, ct])
 
+def symmat6(a_xx, a_yy, a_zz, a_xy, a_xz, a_yz):
+    '''
+    modified from pyrocko
+    Create symmetric 3x3 matrix from its 6 non-redundant values.
+    '''
+    return num.matrix([[a_xx, a_xy, a_xz],
+                       [a_xy, a_yy, a_yz],
+                       [a_xz, a_yz, a_zz]], dtype=num.float)
+
 def pol_synthetics(sources, targets):  
     takeoffangles = [target.take_offangle_rad for target in targets]
     azimuths = [target.azimuth_rad for target in targets]
     gamma_co = gamma(takeoffangles, azimuths)
-    amps = num.array([num.diag(gamma_co.T.dot(sources[0].m9/sources[0].moment).dot(gamma_co))]).T
```

Comment:
need to go over the source.discretize_basesource method here to get sources that are then expressed in the form of many MTs if sources have an extent ...